pkg/controller/stats/controller.go

```diff
@@ -0,0 +1,227 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stats
+
+import (
+	"database/sql"
+	"fmt"
+
+	"k8s.io/client-go/kubernetes"
+
+	"antrea.io/theia/pkg/util"
+)
+
+type diskInfo struct {
+	shard          string
+	name           string
+	path           string
+	freeSpace      string
+	totalSpace     string
+	usedPercentage string
+}
+
+type tableInfo struct {
+	shard      string
+	database   string
+	tableName  string
+	totalRows  sql.NullString
+	totalBytes sql.NullString
+	totalCols  string
+}
+
+type insertRate struct {
+	shard       string
+	rowsPerSec  string
+	bytesPerSec string
+}
+
+type stackTraces struct {
+	shard          string
+	traceFunctions string
+	count          string
+}
+
+const (
+	diskQuery int = iota
+	tableInfoQuery
+	// average writing rate for all tables per second
+	insertRateQuery
+	stackTracesQuery
+)
+
+var queryMap = map[int]string{
+	diskQuery: `
+SELECT
+	shardNum() as Shard,
+	name as DatabaseName,
+	path as Path,
+	formatReadableSize(free_space) as Free,
+	formatReadableSize(total_space) as Total,
+	TRUNCATE((1 - free_space/total_space) * 100, 2) as Used_Percentage
+FROM cluster('{cluster}', system.disks);`,
+	tableInfoQuery: `
+SELECT
+	Shard,
+	DatabaseName,
+	TableName,
+	TotalRows,
+	TotalBytes,
+	TotalCols
+FROM (
+	SELECT
+		shardNum() as Shard,
+		database AS DatabaseName,
+		name AS TableName,
+		total_rows AS TotalRows,
+		formatReadableSize(total_bytes) AS TotalBytes
+	FROM cluster('{cluster}', system.tables) WHERE database = 'default'
+	) as t1
+	INNER JOIN (
+		SELECT
+			shardNum() as Shard,
+			table_catalog as DatabaseName,
+			table_name as TableName,
+			COUNT(*) as TotalCols
+		FROM cluster('{cluster}', INFORMATION_SCHEMA.COLUMNS)
+		WHERE table_catalog == 'default'
+		GROUP BY table_name, table_catalog, Shard
+		) as t2
+		ON t1.DatabaseName = t2.DatabaseName and t1.TableName = t2.TableName and t1.Shard = t2.Shard`,
+	// average writing rate for all tables per second
+	insertRateQuery: `
+SELECT
+	sd.Shard,
+	sd.RowsPerSecond,
+	sd.BytesPerSecond
+FROM (
+	SELECT
+		shardNum() as Shard,
+		(intDiv(toUInt32(date_trunc('minute', toDateTime(event_time))), 2) * 2) * 1000 as t,
+		TRUNCATE(avg(ProfileEvent_InsertedRows),0) as RowsPerSecond,
+		formatReadableSize(avg(ProfileEvent_InsertedBytes)) as BytesPerSecond,
+		ROW_NUMBER() OVER(PARTITION BY shardNum() ORDER BY t DESC) rowNumber
+	FROM cluster('{cluster}', system.metric_log)
+	GROUP BY t, shardNum()
+	ORDER BY t DESC, shardNum()
+	) sd
+WHERE sd.rowNumber=1`,
+	stackTracesQuery: `
+SELECT
+	shardNum() as Shard,
+	arrayStringConcat(arrayMap(x -> demangle(addressToSymbol(x)), trace), '\\n') AS trace_function,
+	count()
+FROM cluster('{cluster}', system.stack_trace)
+GROUP BY trace_function, Shard
+ORDER BY count()
+DESC SETTINGS allow_introspection_functions=1`,
+}
+
+type StatsController struct {
```

Comment:
Naming of "controller" hints that there's some long-running routines involved, which is not the case here. I think we can simply call it ClickHouseStatQuerier or add "impl" suffix to distinguish from the interface. filename could also be renamed and mention clickhouse in case we'd want to collect stat from other components in the future.